src/processNonPendingStatus.ts

```diff
@@ -41,49 +33,34 @@ export async function processNonPendingStatus(
     // Commit that trigger this hook is not the latest commit of the merging PR
     return
   }
-  const baseBranchRule = branchProtectionRules.nodes.find(
-    (rule) => rule.pattern === mergingPr.baseRef.name
-  )
-  if (!baseBranchRule) {
-    // TODO: No protection rule for merging this PR. Merge immediately?
-    return
-  }
-  const requiredCheckNames = baseBranchRule.requiredStatusCheckContexts
+  const checksToSkip: string = process.env.INPUT_CHECKS || ""
+  const checksToSkipList = checksToSkip.split(",")
 
   if (state === "success") {
-    const isAllRequiredCheckPassed = requiredCheckNames.every((checkName) => {
-      if (!checkName.includes("ci/circleci")) {
-        // TODO: Support GitHub Action. Can't get `statusCheckRollup` to work in GitHub API Explorer for some reason.
-        return true
+    const isAllRequiredCheckPassed = latestCommit.checkSuites.nodes.every(
+      (node) => {
+        let status = node.checkRuns.nodes[0]?.status
+        if (node.checkRuns.nodes[0]?.name in checksToSkipList) {
+          status = "COMPLETED"
+        }
+        return status === "COMPLETED" || status === null || status === undefined
```

Comment:
nit:
you can use js's .filter to filter, and .map to map
```suggestion
    const isAllRequiredCheckPassed = latestCommit.checkSuites.nodes.filter(
    node => !(node.checkRuns.nodes[0]?.name in checksToSkipList)
    ).map(
      (node) => {
        let status = node.checkRuns.nodes[0]?.status
        return status === "COMPLETED" || status === null || status === undefined
```